src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -859,7 +873,7 @@ void grpc_chttp2_encode_header(grpc_chttp2_hpack_compressor* c,
         (static_index =
              reinterpret_cast<grpc_core::StaticMetadata*>(GRPC_MDELEM_DATA(md))
                  ->StaticIndex()) < GRPC_CHTTP2_LAST_STATIC_ENTRY) {
-      emit_indexed(c, static_cast<uint32_t>(static_index + 1), &st);
+      emit_indexed<true>(c, static_cast<uint32_t>(static_index + 1), &st);
```

Comment:
Are our tests good enough to catch any bugs in any assumptions like these?

Reply:
The assumption here is safe; true means that the metadata is static, which we established as a necessary condition to enter this if branch.

To make it safer, we could add a debug assert in emit_indexed() that ensures the metadata is indeed static when <true> is specified.